batch7crud-roll8/src/main/java/com/lftechnology/batch7crud/services/EmployeeService.java

```diff
@@ -2,19 +2,34 @@
 
 import com.lftechnology.batch7crud.dao.EmployeeDao;
 import com.lftechnology.batch7crud.exception.DataException;
+import com.lftechnology.batch7crud.exception.ValidationException;
 import com.lftechnology.batch7crud.model.Employee;
+import com.lftechnology.batch7crud.validator.EmployeeValidator;
 
 import java.util.List;
+import java.util.Map;
 
 /**
  * Created by Grishma Shrestha grishmashrestha@lftechnology.com on 1/19/16.
  */
 
 public class EmployeeService {
-  private EmployeeDao employeeDao = new EmployeeDao();
+  private EmployeeDao employeeDao;
+  private EmployeeValidator validator;
 
-  public Employee create(Employee employee) throws DataException {
-    return employeeDao.create(employee);
+  public EmployeeService() {
+    employeeDao = new EmployeeDao();
+    validator = new EmployeeValidator();
+  }
+
+  public Employee create(Employee employee) throws DataException, ValidationException { // NOSONAR
+    Map<String, String> errors= validator.validate(employee);
```

Comment:
employee should be validated at controller level. You are validating at service layer which is 1 more extra layer to reach out.